### ISSUE TITLE
GFF2XML -> XML2GFF (invalid base64 length)

### ISSUE THREAD

**Issue by drake127** [xoreos/xoreos-tools#67]
Hi,

it looks like xml2gff doesn't work for XMLs that contains longer base64 as it fails because of "Invalid length for a base64-encoded string".

I checked the string itself and it's correct, just contains whitespaces from XML formatting (and newlines).

**Comment by DrMcCoy** [comment]
Can you give me an example file to check against?

**Comment by drake127** [comment]
Sure, here you are. Only thing I did differently I used --encoding 0=cp-1250 but I don't think it should matter in this case.

It fails in countLength(Ustring) with string:
```

      RHJvYm7saprtIHphaGFsZW7hIHBvc3RhdmEgdiBwcm9zdP1jaCBsZXNu7WNoIJph
      dGVjaCBzIG1l6GVtIHUgcGFzdS4KCg==

```

(url)
[serialized.zip](https://github.com/xoreos/xoreos-tools/files/5622919/serialized.zip)

**Comment by DrMcCoy** [comment]
Thanks, I'll have a look at it later in the evening

**Comment by drake127** [comment]
FYI - I modified decodeBase64 to strip whitespace characters and it is working now.

Just one unrelated question though, the resulting GFF is 5 KB less than the original. When I convert it back to XML, the result is the same. Is it expected? The header looks entirely different:
![image](https://user-images.githubusercontent.com/8018723/100756126-34246c00-33ed-11eb-971c-506a79fc21bd.png)

**Comment by DrMcCoy** [comment]
Ah, yes, found the issue. Should be fixed with c6315f75a8a366beb898718a098aa29cc22f70ac, thanks for reporting! :)

Yeah, the file being shorter is okay. Our GFF code tries harder to consolidate the same string data (*). For example, the original files contains the string "STAMINA_MAX" multiple times, while the GFF produced by xml2gff contains the string just once, and all fields using that same string just reference this one instance.

That also explains the differences in the header, because those values there are offsets to the different sections in the GFF, one of them being the string table. We're creating files that are logically identical, i.e. that contain the exact same information, not files that are byte-by-byte identical.

(*) Technically, it just throws extended field data value into a map and duplicates get consolidated that way.
